worlds/pokemon_rb/client.py

```diff
@@ -0,0 +1,277 @@
+import base64
+import logging
+import time
+
+from NetUtils import ClientStatus
+from worlds._bizhawk.client import BizHawkClient
+from worlds._bizhawk import read, write, guarded_write
+
+from worlds.pokemon_rb.locations import location_data
+
+logger = logging.getLogger("Client")
+
+BANK_EXCHANGE_RATE = 100000000
+
+DATA_LOCATIONS = {
+    "ItemIndex": (0x1A6E, 0x02),
+    "Deathlink": (0x00FD, 0x01),
+    "APItem": (0x00FF, 0x01),
+    "EventFlag": (0x1735, 0x140),
+    "Missable": (0x161A, 0x20),
+    "Hidden": (0x16DE, 0x0E),
+    "Rod": (0x1716, 0x01),
+    "DexSanityFlag": (0x1A71, 19),
+    "GameStatus": (0x1A84, 0x01),
+    "Money": (0x141F, 3),
+    "ResetCheck": (0x0100, 4),
+    # First and second Vermilion Gym trash can selection. Second is not used, so should always be 0.
+    # First should never be above 0x0F. This is just before Event Flags.
+    "CrashCheck1": (0x1731, 2),
+    # Unused, should always be 0. This is just before Missables flags.
+    "CrashCheck2": (0x1617, 1),
+    # Progressive keys, should never be above 10. Just before Dexsanity flags.
+    "CrashCheck3": (0x1A70, 1),
+    # Route 18 script value. Should never be above 2. Just before Hidden items flags.
+    "CrashCheck4": (0x16DD, 1),
+}
+
+location_map = {"Rod": {}, "EventFlag": {}, "Missable": {}, "Hidden": {}, "list": {}, "DexSanityFlag": {}}
+location_bytes_bits = {}
+for location in location_data:
+    if location.ram_address is not None:
+        if type(location.ram_address) == list:
+            location_map[type(location.ram_address).__name__][(location.ram_address[0].flag, location.ram_address[1].flag)] = location.address
+            location_bytes_bits[location.address] = [{'byte': location.ram_address[0].byte, 'bit': location.ram_address[0].bit},
```

Comment:
Still a bunch of single-quoted strings here and there that should be double-quoted